src/rendering/renderers/pool/PoolLockRenderer.ts

```diff
@@ -0,0 +1,171 @@
+import config from '../../../config'
+import { IGraphicStorage, RenderingContext } from '../..'
+import { BaseRenderer, GraphicUtils } from '../..'
+
+import datamath from '../../../lib/datamath'
+import { Graphics, Container, Text, Sprite } from '../../../lib/pixi'
+import { LOCK_ICON_TEXTURE } from '../../textures/symbols'
+export class PoolLockRenderer extends BaseRenderer {
+
+    static readonly POOL_LOCK_ID: symbol = Symbol('POOL_LOCK_ID')
+
+    private lineStyle: any
+
+    private torusStyle: any
+
+    private iconStyle: any
+
+    private coverStyle: any
+
+    constructor(renderer: IGraphicStorage) {
+        super(renderer)
+
+        this.coverStyle = {
+            paddingx: 7,
+            paddingy: 5,
+
+            paddingTop: 30,
+            paddingLeft: 10,
+
+            radius: 8,
+            color: 0xFFA000,
+        }
+
+        this.iconStyle = {
+            size: 13,
+        }
+
+        this.torusStyle = {
+            innerr: 2.5,
+            outerr: 5,
+            paddingTop: 10,
+            color: 0xFFA000,
+        }
+
+        this.lineStyle = {
+            width: 2,
+            join: 'round',
+            cap: 'round',
+            gap: 10,
+            dash: 10,
+            paddingTop: 5,
+            paddingBottom: 20,
+            color: 0xFFA000,
+        }
+    }
+
+    public get rendererId(): symbol {
+        return PoolLockRenderer.POOL_LOCK_ID
+    }
+
+    protected update(
+        context: RenderingContext,
+        container: Container,
+    ): Container {
+        if (!context.pool) {
+            container.alpha = 0
+
+            return container
+        }
+
+        container.alpha = 1
+
+        this.updatePoolBorder(context, container)
+
+        return container
+    }
+
+    private updatePoolBorder(context: RenderingContext, container: Container): Container {
+        const {
+            width,
+            height
+        } = context.screen
+
+        const { timerange } = context.plotdata
+        const [x] = datamath.scale([context.pool.lockDate], timerange, width)
+
+        const [cover, coverstate] = this.get<Container>('poolIcon', () => this.createPoolIcon(context))
+        if (coverstate.new) {
+
+            cover.position.y = this.coverStyle.paddingTop
```

Comment:
You should update position x and y on each render
Because scale of the screen could change